contrib_bots/lib/converter.py

```diff
@@ -0,0 +1,228 @@
+               various units, e.g. Celsius to Fahrenheit,
+               or Kilobytes to Gigabytes. It looks for messages of
+               the format '@convert <number> <original-unit> <new-unit>.
+               The message '@convert help' posts a short description of
+               how to use the plugin, along with a list of
+               all supported units.
+               '''
+
+    def triage_message(self, message, client):
+        return ('@convert' in message['content'])
+
+    def handle_message(self, message, client, state_handler):
+        content = message['content']
+
+        words = content.lower().split()
+        conversion_occ = [idx for idx, x in enumerate(copy.copy(words)) if x == "@convert"]
+        results = []
+
+        for idx in conversion_occ:
+            if (idx + 1) < len(words) and words[idx + 1] == 'help':
+                results.append('Converter usage:\n' +
+                               '`@convert <number> <unit_from> <unit_to>`\n' +
+                               'Converts `number` in the unit `unit_from` to ' +
+                               'the `unit_to` and prints the result\n' +
+                               '`number`: floating point number, e.g. 12, 13.05, 0.002\n' +
+                               '`unit_from` and `unit_to` are two of the following units:\n' +
+                               '* Bit, byte\n' +
+                               '* Kelvin (K), Fahrenheit (F), Celsius(C)\n' +
+                               '* Meter (m), Inch (in), Foot (ft), Yard (y),' +
+                               ' Mile (mi), Nautical-Mile (nmi)\n' +
+                               '* Square-Meter (m^2), Square-Kilometer (km^2),' +
+                               'Square-Centimeter (cm^2), Are, Hectare (ha),' +
+                               ' Square-Inch (in^2), Square-Foot (ft^2), Square-Yard (y^2), ' +
+                               'Acre, Square-Mile(mi^2)\n' +
+                               '* Cubic-Meter (m^3), Cubic-Centimeter (cm^3), Cubic-Decimeter, ' +
+                               'Cubic-Inch,' +
+                               ' Cubic-Foot (ft^3), Cubic-Yard (y^3), Fluid-Ounce, Liter (l) (\n' +
+                               '* Gram (g), Ton (t), Ounce (oz), Pound(lb)\n\n\n' +
+                               'Allowed prefixes are:\n' +
+                               '* atto, pico, femto, nano, micro, milli, centi, deci\n' +
+                               '* deca, hecto, kilo, mega, giga, tera, peta, exa\n\n\n' +
+                               'Usage examples:\n' +
+                               '* `@convert 12 celsius fahrenheit`\n' +
+                               '* `@convert 0.002 kilomile millimeter`\n' +
+                               '* `@convert 31000 attocubic-decimeter megaounce`\n' +
+                               '* `@convert 56 g lb`\n'
+                               )
+                continue
+            if (idx+3) < len(words):
```

Comment:
I think there are other places as well where you need to space around the operator.